fiftyone/operators/panel.py

```diff
@@ -118,39 +141,38 @@ class WriteOnlyError(Exception):
     """Error raised when trying to read a write-only property."""
 
 
-class PanelRefBase:
-    """
-    Base class for panel state and data.
+class PanelRefBase(object):
+    """Base class for panel state and data.
 
-    Attributes:
-        _data (dict): A dictionary to store the data or state.
-        _ctx: The context object containing the operations.
+    Args:
+        ctx: an :class:`fiftyone.operators.executor.ExecutionContext`
     """
 
     def __init__(self, ctx):
         self._data = {}
         self._ctx = ctx
 
-    def set(self, key, value):
-        """
-        Sets the value in the dictionary.
+    def set(self, key, value=None):
+        """Sets some value(s) in the dictionary.
 
         Args:
-            key (str): The key.
-            value (any): The value.
+            key: a key, ``"nested.key.path"``, or dict mapping multiple
+                possibly-nested keys to values
+            value (None): the value, if key is a string
         """
-        pydash.set_(self._data, key, value)
+        d = {key: value} if value is not None else key
+        for k, v in d.items():
+            pydash.set_(self._data, k, v)
 
     def get(self, key, default=None):
-        """
-        Gets the value from the dictionary.
+        """Gets a value from the dictionary.
 
         Args:
-            key (str): The key.
-            default (any): The default value if key is not found.
+            key: a key
```

Comment:
This should be the same description as the others. And we should match [the pydash description](https://pydash.readthedocs.io/en/latest/deeppath.html).

> A deep path string is used to access a nested data structure of arbitrary length. Each level is separated by a "." and can be used on both dictionaries and lists. If a "." is contained in one of the dictionary keys, then it can be escaped using "\". For accessing a dictionary key that is a number, it can be wrapped in brackets like "[1]".

Reply:
If theres a way to define this once and refer to it for all `key` params... that would be ideal.

Reply:
Let's not encourage people to use numeric keys or keys that contain `.`. That will break other things like the ability to use attribute access syntax `ctx.panel.state.key`.

Will update this `key` docstring to match the others though 👍